src/study/system-model/include/antares/study/system-model/library.h

```diff
@@ -79,6 +79,7 @@ class LibraryBuilder
     LibraryBuilder& withDescription(const std::string& description);
     LibraryBuilder& withPortTypes(std::vector<PortType>&& portTypes);
     LibraryBuilder& withModels(std::vector<Model>&& models);
+    LibraryBuilder& withModels2(std::unordered_map<std::string, Model>&& models);
```

Comment:
While I understand why you introduce a 2nd method, I think you could do without it (using existing method `withModels`)